icicle/backend/cpu/src/hash/cpu_merkle_tree.cpp

```diff
@@ -40,34 +40,42 @@ namespace icicle {
 
         if (0 < layer_idx) {
           // update nof_hashes to the next layer (below)
-          const uint64_t cur_layer_input_size = layer_hashes[layer_idx].default_input_chunk_size();
-          nof_hashes = nof_hashes * cur_layer_input_size / layer_hashes[layer_idx - 1].output_size();
+          const uint64_t next_layer_total_size = layer_hashes[layer_idx - 1].output_size();
```

Comment:
did you mean to edit this file? Maybe you are not merged with main?

Reply:
No
It's not me

Reply:
Why did you resolve? It's an issue